CRM/Contact/BAO/Query.php

```diff
@@ -1489,6 +1489,15 @@ public static function convertFormValues(&$formValues, $wildcard = 0, $useEquals
           $params[] = array('on_hold', '=', $formValues['email_on_hold']['on_hold'], 0, 0);
         }
       }
+      elseif (substr($id, 0, 7) == 'custom_'
+        &&  (
+          substr($id, -9, 9) == '_relative'
+          || substr($id, -5, 5) == '_from'
+          || substr($id, -3, 3) == '_to'
+        )
+      ) {
+        self::convertCustomRelativeFields($formValues, $params, $values, $id);
+      }
```

Comment:
The missing piece to support non-range custom date fields is to add

      elseif (!empty($values) && !is_array($values) && self::isCustomDateField($id) ) {
        $params[] = array($id, '=', CRM_Utils_Date::processDate($values), 0, $wildcard);
      }

In 4.7 the date is already formatted to a consistent date format by the time it hits this point by the datepicker replacement.